source/agora/script/Opcodes.d

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+
+    Contains the supported opcodes for the basic execution engine (non-webASM)
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.script.Opcodes;
+
+import std.traits : EnumMembers;
+
+/*******************************************************************************
+
+    The supported opcodes for the script execution engine.
+
+    Opcodes named `CHECK_*` push their result to the stack,
+    whereas `VERIFY_*` opcodes will cause the script execution to fail.
+
+    We can encode up to 255 opcodes (one of which is INVALID to halt execution).
+
+    Note that the range of values between `PUSH_BYTES_1` and `PUSH_BYTES_64`
+    encodes a push to the stack between 1 .. 64 bytes. This range was chosen
+    to allow pushing one of our largest types (Hash) in a single opcode,
+    but is small enough to leave room for introducing other 1-byte opcodes.
+
+    For pushes of data longer than 64 bytes use one of the `PUSH_DATA_*`
+    opcodes.
+
+*******************************************************************************/
+
+public enum OP : ubyte
+{
+    /// Executing this is an error and will cause script execution to fail.
+    /// Purposefully located first to default OPs to errors.
+    INVALID = 0x45,
+
+    /// Used to encode a small data push to the stack (up to 64 bytes),
+    /// may be used with `case PUSH_BYTES_1: .. case PUSH_BYTES_64:` syntax.
+    PUSH_BYTES_1 = 0x01,
+
+    /// Ditto
+    PUSH_BYTES_64 = 0x40, // 64 decimal
```

Comment:
This will need to be extended. If we want to support `SIGHASH` then we need to encode it after the signature. So that will make needing 65-byte pushes common.

Let's be on the safe side and allow up to 75-byte pushes.